osgp/platform/osgp-adapter-domain-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/domain/da/application/services/BaseService.java

```diff
@@ -74,6 +74,10 @@ protected Organisation findOrganisation(final String organisationIdentification)
         return organisation;
     }
 
+    protected RtuDevice findRtuDeviceForDevice(final Device device) {
+        return this.rtuDeviceRepository.findById(device.getId()).get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS3655)

Reply:
Made method return an Optional